bpf-sys/src/uname.rs

```diff
@@ -25,13 +25,18 @@ pub fn uname() -> Result<::libc::utsname, ()> {
 
 #[inline]
 pub fn get_kernel_internal_version() -> Option<u32> {
-    let version = if let Ok(version) = fs::read_to_string("/proc/version_signature") {
-        parse_version_signature(&version.trim())?
-    } else {
-        to_str(&uname().ok()?.release).into()
+    let mut version = None;
+
+    if let Ok(version_signature) = fs::read_to_string("/proc/version_signature") {
+        version = parse_version_signature(&version_signature.trim());
+    }
```

Comment:
Since you're using a separate var for `final_version` (which is a cleaner way!), I think `version` doesn't have to be a mutable var:
```rust
    let version = if let Ok(version_signature) = fs::read_to_string("/proc/version_signature") {
        parse_version_signature(&version_signature.trim())
    } else {
        None
    };
```
Or, even simpler:
```suggestion
    let version = fs::read_to_string("/proc/version_signature").ok()
        .and_then(|version_signature| parse_version_signature(&version_signature.trim()));
```